src/riak_ql_ddl_util.erl

```diff
@@ -0,0 +1,122 @@
+%% -------------------------------------------------------------------
+%%
+%% riak_ql_ddl_util: utility functions for use in the riak ddl helper
+%%                   modules generated by the riak_ql_ddl_compiler
+%%
+%% Copyright (c) 2016 Basho Technologies, Inc.  All Rights Reserved.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+-module(riak_ql_ddl_util).
+
+-export([
+         is_valid_delete_where_clause/1,
+         make_delete_key/2
+        ]).
+
+-include("riak_ql_ddl.hrl").
+
+-define(EMPTY_ERRORS, []).
+-define(EMPTY_KVs,    []).
+
+make_delete_key(#key_v1{ast = AST}, Where) ->
+    KVs = make_KV(Where, ?EMPTY_KVs),
+    Key = [proplists:get_value(K, KVs) || ?SQL_PARAM{name = [K]} <- AST],
+    {ok, Key}.
+
+is_valid_delete_where_clause(Where) ->
+    case is_valid2(Where, ?EMPTY_ERRORS) of
+        ?EMPTY_ERRORS -> true;
```

Comment:
How does the validation deal with queries with insufficient values? As I read the code, `make_delete_key` will return `undefined` for each key field not mentioned in `'WHERE'`, which would lead to deleting.. what?